archunit-junit/junit5/engine/src/main/java/com/tngtech/archunit/junit/DisplayNameResolver.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2014-2021 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.junit;
+
+import com.tngtech.archunit.ArchConfiguration;
+
+final class DisplayNameResolver {
+
+    static final String DISPLAYNAME_REPLACE_UNDERSCORES_PROPERTY_NAME = "displayName.replaceUnderscores";
+
+    static String determineDisplayName(String elementName) {
+        if (replaceUnderscores()) {
+            return underscoresReplacedBySpaces(elementName);
+        } else {
+            return elementName;
+        }
+    }
+
+    private static String underscoresReplacedBySpaces(String elementName) {
```

Comment:
Should we make the property also `replaceUnderscoresBySpaces`? I like this method name, because it says exactly what it does. The property name just says `replaceUnderscores`, but not with what :thinking: